### PR TITLE
[Impeller] Add debug captures and inspector.

### PR DESCRIPTION
**Pull request by bdero**
Weekend project! Press `C` to capture in the Aiks playground.

Decided to finally give this a go and attempt to relieve some of the print debugging/mindfuck around investigation of coverage-related issues lately. :)

Captures:
* Capture documents from anywhere in Impeller.
* Easily implement inspectors for those documents.
* Replay documents with live editing.
* No overhead when capturing is build time disabled (that's the idea, anyway).
* Low overhead when capturing is runtime disabled.

Aiks inspector:
* Outline passes and rendered entities.
* Identify collapsed passes.
* Visibly highlight coverage.
* Live edit scene properties.

Possible future work:
* Filters!
* Blend mode property.
* Pointer + release proc property.
* Support captures in the DL playground.
* Text atlas visualization.
* Multi-frame capture and scrubbing.
* Menus instead of key bindings?

https://github.com/flutter/engine/assets/919017/a7a63e24-f72f-4140-a21e-6ca02a05fc20

### CODE REVIEW

**Comment by chinmaygarde** [comment]
whoa

**Review by chinmaygarde — APPROVED** [review]
Sorry about the really late review. I was on the fence about the intrusiveness of this but can definitely see the value. Especially as we build on top it. I think we should land and iterate on it and keep evaluating this from time to time.

File: impeller/entity/contents/solid_color_contents.cc [hunk 1]
```diff
@@ -44,6 +44,8 @@ std::optional<Rect> SolidColorContents::GetCoverage(
 bool SolidColorContents::Render(const ContentContext& renderer,
                                 const Entity& entity,
                                 RenderPass& pass) const {
+  auto capture = entity.GetCapture().CreateChild("SolidColorContents");
```
Comment (jonahwilliams) [inline impeller/entity/contents/solid_color_contents.cc:47]: Can we IFDEF these out?

Comment (bdero) [reply impeller/entity/contents/solid_color_contents.cc:47]: There's theoretically no need. All this stuff is inlined and should flatten out to zero instructions when the capture is build-time disabled. (I haven't rigorously proven this, mind you)

Comment (jonahwilliams) [reply impeller/entity/contents/solid_color_contents.cc:47]: Well, I'll see it in the profile if its there  😈  - but yeah, no need to worry about it until its a problem !

Comment (bdero) [reply impeller/entity/contents/solid_color_contents.cc:47]: I'm gonna Volkswagen the heck out of our benchmarks, but in a benign way.

Comment (jonahwilliams) [reply impeller/entity/contents/solid_color_contents.cc:47]: Just add some speedup loops

**Pull request merged**